### ISSUE TITLE
0.83.16 MinGW SDL2 32bit can't run in windows xp

### ISSUE THREAD

**Issue by leoxxx** [joncampbell123/dosbox-x#2752]
A dialog prompt: The program input point GetFileInformbyHandleEx cann't be located on the dynamic link library KERNEL32 .dll.
0.83.15 is O.K.

**Comment by joncampbell123** [comment]
That function was added in Windows Vista: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getfileinformationbyhandleex

If Windows XP compatibility is needed, then that function should be called through a function pointer obtained by GetProcAddress(GetModuleHandle("KERNEL32.DLL"),"GetFileInformationByHandleEx")

**Comment by joncampbell123** [comment]
Actually it seems the issue could be resolved if MinGW builds did not define "PHYSFS_PLATFORM_WINRT". "WINRT" should only ever be considered if targeting Windows Vista or higher.

**Comment by joncampbell123** [comment]
physfs platform determination is in src/libs/physfs/physfs_platforms.h

It seems the best case is to add !defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) to the conditional check leading to WINRT support.

**Comment by Wengier** [comment]
FYI: Before new MinGW builds are posted you can also try the patched 32-bit Visual Studio builds included in the installer.

**Comment by joncampbell123** [comment]
It turns out the code that calls that function is in libglib-2.0, which is a dependency of *drumroll* libslirp.

Sorry @Wengier, but if MinGW builds are to support Windows XP, they're going to have to omit libslirp support.

**Comment by Wengier** [comment]
@Joncampbell123 I tried to limit libslirp support to 64-bit non-"low-end" builds earlier (I suppose mingw-w64-x86_64-libslirp is 64-bit package whereas mingw-w64-i686-libslirp is 32-bit package) so that Windows XP will be supported. Not sure why Slirp is enabled in 32-bit MinGW as well. Perhaps you installed mingw-w64-i686-libslirp too?

**Comment by joncampbell123** [comment]
I see, I'll try again.

**Comment by leoxxx** [comment]
I keep 0.83.15 for XP

**Comment by joncampbell123** [comment]
I'm seeing a problem here where even if I fix that function call not to be made, something else linked to in the latest MinGW then calls "GetTickCount64" which is also not in XP. It's getting difficult to use the latest MinGW to make Windows XP compatible binaries.

**Comment by Wengier** [comment]
So I guess patched 32-bit Visual Studio builds may in fact be the easiest solution at this time. They will have the debugger too which does not exist in MinGW builds.

@leoxxx You can just get the 32-bit Visual Studio builds of 0.83.16 to run in XP.

**Comment by DosFreak** [comment]
joncampbell123,

Possibly:
https://github.com/msys2/MINGW-packages/issues/5139

I compile without posix so don't have this issue. Last time I bothered with pacman with MSYS2 it required POSIX so I had to copy over dwarf win32 mingw-w64 build for NT3.51+ support.

So for whichever environment you are compiling DOSBox in check to see if threads are win32 or posix.

May or may not be helpful, need to see about updating since last time I updated my compile environment was jan 2020:
https://www.vogons.org/viewtopic.php?p=827546#p827546
